src/malloc/mallocng/meta.h

```diff
@@ -6,22 +6,47 @@
 #include <limits.h>
 #include "glue.h"
 
+static int get_errno()
+{
+	return errno;
+}
+
+static void set_errno(int e)
+{
+	errno = e;
+}
+
 __attribute__((__visibility__("hidden")))
+#ifdef __CHEERP__
+__attribute__((cheerp_asmjs))
+#endif
 extern const uint16_t size_classes[];
 
 #define MMAP_THRESHOLD 131052
 
 #define UNIT 16
 #define IB 4
 
-struct group {
-	struct meta *meta;
-	unsigned char active_idx:5;
+struct
+#ifdef __CHEERP__
+__attribute__((cheerp_asmjs))
+#endif
+group {
+	struct
+#ifdef __CHEERP__
```

Comment:
I *think* that the attribute in the inner struct is not required, since the enclosing context is the outer struct with the attribute. could you try if it's the case?

Reply:
I think because there's no forward declaration, it's seeing this as the declaration. It does not compile without the attribute here.

Reply:
ok